OTAnalytics/domain/section.py

```diff
@@ -41,28 +42,16 @@ def notify_sections(self, sections: list[SectionId]) -> None:
         pass
 
 
-VALUE = TypeVar("VALUE")
-
-
 SectionChangedObserver = Callable[[SectionId], None]
 
 
-class SectionChangedSubject(Generic[VALUE]):
+class SectionChangedSubject(Registrable[SectionChangedObserver]):
     """
     Helper class to handle and notify observers
     """
 
     def __init__(self) -> None:
-        self.observers: set[SectionChangedObserver] = set()
-
-    def register(self, observer: SectionChangedObserver) -> None:
-        """
-        Listen to events.
-
-        Args:
-            observer (SectionChangedObserver): listener to add
-        """
-        self.observers.add(observer)
+        super().__init__()
 
     def notify(self, value: SectionId) -> None:
```

Comment:
Can we also move the notify method into the super class with a second generic type?

This would replace a lot of `...Subject` classes with just one.